packages/devextreme/js/__internal/grids/new/card_view/context_menu/controller.ts

```diff
@@ -65,32 +72,55 @@ export class ContextMenuController
   }
 
   private getSortingItems(column: Column): ContextMenuItem[] {
-    const onItemClick = (e: ItemClickEvent): void => {
-      this.columnsController.columnOption(column, 'sortOrder', e.itemData?.value);
-    };
+    const mode = this.sortingController.mode.value;
+    const isDisabled = mode === 'none' || !column.allowSorting;
 
     return [
       {
         text: this.options.oneWay('sorting.ascendingText').peek(),
         value: 'asc',
-        disabled: column.sortOrder === 'asc',
+        disabled: isDisabled || column.sortOrder === 'asc',
         icon: 'sortuptext',
-        onItemClick,
+        onItemClick: (event: ContextMenuItemPressedEvent): void => {
+          this.handleSortMenuClick(event, mode, column);
+        },
```

Comment:
can we move this function to a variable and not duplicate it for each item?


        
```suggestion
const onItemClick = (event: ContextMenuItemPressedEvent): void => {
    =this.handleSortMenuClick(event, mode, column);
},

return [{
        // ...,
        onItemClick,
        }, {
        // ...,
        onItemClick
        }];
```

Reply:
I guess it's a minor suggestion, so @Raushen it's up to u :)